addons/caches/functions/fnc_moduleInitCaches.sqf

```diff
@@ -0,0 +1,48 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Creates specified number of caches
+
+    Parameters:
+        0: The module object <OBJECT>
+
+    Return value:
+        Nothing
+*/
+
+(_this select 1) params ["_module"];
+
+if (!isServer) exitWith {};
+
+private _numberOfCaches = _module getVariable [QGVAR(cacheNumber), 0];
+private _cacheTypes = parseSimpleArray (_module getVariable [QGVAR(cachePoolString), ""]);
+private _contentTypes = parseSimpleArray (_module getVariable [QGVAR(cacheContentPoolString), ""]);
+private _area = _module getVariable ["objectarea", []];
+
+for "_i" from 1 to _numberOfCaches do {
+
+    // get a position
+    _area deleteAt 4;
+    private _areaArray = [(getPos _module)] + _area;
+    private _position = [_areaArray] call CBA_fnc_randPosArea;
+    private _nearestCover = (nearestTerrainObjects [_position, ["TREE", "SMALL TREE", "HOUSE"], 100]) select 0;
+
+    // set nearestCover to the random pos if its empty
+    if (_nearestCover isEqualTo []) then {
```

Comment:
If `nearestTerrainObjects` returns nothing, what does `select 0` over that do?
Meaning, is the check against an empty array here correct (and tested)?

Reply:
Added an else clause to the if statement. The select 0 is now part of the if statement.